jenkins/ml-models.JenkinsFile

```
@@ -28,17 +28,25 @@ pipeline {
         )
         }
     environment {
-        ARTIFACT_PATH = "${BASE_DOWNLOAD_PATH}/${VERSION}/"
+        ARTIFACT_PATH = "${env.BASE_DOWNLOAD_PATH == "ml-models/model_listing" ? "${BASE_DOWNLOAD_PATH}" : "${BASE_DOWNLOAD_PATH}/${VERSION}/"}"
         UPLOAD_PATH = "models/ml-models"
         BUCKET_NAME = credentials('ml-models-bucket-name')
     }
     stages{
         stage('Parameters Check') {
             steps {
                 script {
-                    if(BASE_DOWNLOAD_PATH.isEmpty() || VERSION.isEmpty()) {
+                    if(BASE_DOWNLOAD_PATH == "ml-models/model_listing" && VERSION.isEmpty()) {
+                        echo "Proceeding to download the ml-models/model_listing."
+                    }                  
```

Lots of if and else can we have a better conditions?
Also you can ignore last else block. No need to log normal flow.

```suggestion
                    if(BASE_DOWNLOAD_PATH == "ml-models/model_listing" && !VERSION.isEmpty()){
                        currentBuild.result = 'ABORTED'
                        error('ml-models/model_listing cannot have a version')
                    }                  
```

---

We need all those cases Sayali, removing a condition can cause a run to have a mismatch of parameters.
1) Check for base path as `ml-models/model_listing` and if so version should be empty.
2) Check for empty base path.
3) Check for empty version.
4) This is an else block, to allow move to next stage, hence cant be removed as its an else in if-else condition, not just an if condition.

With the above 4 conditions I have tested the following.
1) With base path as `ml-models/model_listing`, user cant have an version, if version is declared raise an error.
2) User can run the pipeline with an empty base path nor with an empty version value, unless the base path is `ml-models/model_listing`

---

Thanks! Wanted to confirm the 1st point which works I believe. 
My point was the checks are only to abort the builds and do nothing if everything is ok

---

`do nothing if everything is ok`, in an else block i'm forced to move forward and do nothing, hence added an `echo` statement, we cant skip an else block section.

---

Just to add with your suggestion commit, I believe the job passes if user just declared a `VERSION` parameter without `BASE_DOWNLOAD_PATH`, which I have handled in if-else.